### DISCUSSION TITLE
对话和消息删除、重新生成、分支功能有点奇怪

### DISCUSSION THREAD

**Discussion by CookSleep** [CherryHQ/cherry-studio#2025] [General]
# 前言
第一天实际使用Cherry Studio，感觉还挺不错！

最开始我将ChatBox作为主力，后来因为数据同步和函数调用功能需求比较高，转到了Open WebUI。不过比较Open WebUI部署成本还是有的，所以我仍然在我写的国内中转API使用教学中推荐ChatBox，不过我发现Cherry Studio更简洁美观、支持以文本形式上传文件等，美中不足是目前的对话删除、重新生成、分支功能有点奇怪，和其他项目差别有点大，甚至存在一些略显尴尬的部分。

# 反馈
## 对话删除
![image](https://github.com/user-attachments/assets/ac7b82c1-946b-4908-ade5-c7a30451ddff)
当只有一个对话时，我觉得还是允许按叉叉清空对话比较好，这个虽然无伤大雅，但是如果是第一次使用可能会造成一种一时间不知道怎么清理对话的问题（）别的项目是支持这种操作的

## 重新生成、分支
我很多时候并不需要另一个模型给我生成回复，但是目前会默认让我选择一个模型重新生成（或许在设置中加入偏好开关，或者增加悬浮在重新生成上方的使用其他模型的选项？）。

目前重新生成后，历史记录里还“@”那个模型，且旧的记录仍包含在历史记录中……emm……这个感觉没有必要吧

我在这之前见过三种重新生成、分支的方案，第一种就是ChatGPT官网为代表的按下发送后，开启一个可以通过按钮快速切换不同输入和输出的对话分支：
![PixPin_2024-10-14_01-18-48](https://github.com/user-attachments/assets/2003348e-df83-428c-9cef-06f1ce2b9b1f)

第二种是完全不保留旧记录，直接使用新回复覆盖：
![PixPin_2024-10-14_01-20-51](https://github.com/user-attachments/assets/708e8cbe-e7b5-4a50-9d15-d8a68c2bd19c)

第三种是ChatBox那种可以通过点击 `↓` 在下方获取一个新回复、保留旧回复：
![image](https://github.com/user-attachments/assets/0e17f8d8-05a0-4601-b78a-849403e97b78)

我认为第一种最实用（Open WebUI还在这个基础上引入了对话分支预览、点击消息以快速跳转至对应位置的功能）：
![PixPin_2024-10-14_01-28-18](https://github.com/user-attachments/assets/b3495c07-c333-4a1c-abf3-76abf6bc87fd)

如果能实现第一种+Open WebUI用的那种概览图，那真是太棒了！

## 消息删除
我不能理解既然都没有ChatBox那种可以通过点击 `↓` 在下方获取一个新回复的功能，为什么还支持删除单个消息……删除模型回复就会造成下面这种尴尬的局面：
![PixPin_2024-10-14_01-36-20](https://github.com/user-attachments/assets/d7eb90ae-a208-4407-82a5-30aebe5b953d)

---

大概就是这些，希望作者可以考虑一下这些问题，并考虑尽快加入移动端，这样我就可以把Cherry Studio更放心地加入我的教学了！加油！

**Comment by kangfenmao** [comment]
简单的功能可以先改造一下，分叉功能较复杂，先等等，找个整块的时间弄。

**Comment by ZzzzRyan** [comment]
赞同，这个重新生成确实逻辑比较奇怪，主要是还会携带旧的聊天记录，造成对新模型回答的误导。

Open WebUI的对话历史树我也觉得很赞，还可以在多个历史记录中选择，挑选满意的记录引导后续消息的生成。

更进一步，我觉得这个历史记录树的功能在**多模型对话**中应该能发挥更大的作用。比如，多个模型同时生成回答，如果有的模型回答的比较糟糕，我们就可以抛弃它的回答，选择最满意的那一个模型的回答。在下一轮提问中，给所有模型都带上用户认为最好的那个模型的历史回答，让多模型后续对话朝一致、优异的方向进行。

比较希望日后开发多模型对话时，可以考虑融合这个功能。

**Comment by Yanlin-Duanmu** [comment]
被安利了cherry studio，刚刚上手，发现重新生成的功能不太好用，特来反馈，发现已经有人提出了相关issue。

默认的重新生成会新增一个对话同时保留旧的回复，导致原有旧回答污染上下文。避免这个问题，就只能手动删除模型回复。又因为消息删除后不支持让模型再次生成，只好把用户输入也删掉，并把刚刚的问题重新在输入框敲一遍，导致有些繁琐和不人性化。

界面很喜欢，眼前一亮，打算作为未来的主力AI客户端。希望作者能够尽快修复这个体验上的问题呀~

**Comment by kangfenmao** [comment]
> 被安利了cherry studio，刚刚上手，发现重新生成的功能不太好用，特来反馈，发现已经有人提出了相关issue。
> 
> 默认的重新生成会新增一个对话同时保留旧的回复，导致原有旧回答污染上下文。避免这个问题，就只能手动删除模型回复。又因为消息删除后不支持让模型再次生成，只好把用户输入也删掉，并把刚刚的问题重新在输入框敲一遍，导致有些繁琐和不人性化。
> 
> 界面很喜欢，眼前一亮，打算作为未来的主力AI客户端。希望作者能够尽快修复这个体验上的问题呀~

行，我尽快改进一下

**Reply by euzenlee** [reply]
这个功能还没实现吧

**Comment by Zurzark** [comment]
+1，希望优化下这块

**Comment by s97712** [comment]
一些很常用的操作真的非常非常非常非常非常非常非常非常非常繁琐

**Comment by kc0ed** [comment]
请求引入Open WebUI的对话概述，这个真的很管用

**Reply by CookSleep** [reply]
其实好早之前就有了，你把鼠标悬浮在对话框右侧就能看到历史记录图标，点一下就能打开

**Reply by kc0ed** [reply]
> 其实好早之前就有了，你把鼠标悬浮在对话框右侧就能看到历史记录图标，点一下就能打开

cherrystudio无法在同一个话题下创建分支，这个挺令人头疼的。OpenWUI的好处在于分支什么的都可以看到，这个舒服不少

**Comment by jingchang0623-crypto** [comment]
分支功能确实有时候会有点奇怪...我们之前也遇到过类似的问题，后来发现是状态管理的问题。

建议：
1. 尝试清除缓存后重新启动
2. 检查是否有插件冲突
3. 如果还有问题，可以去 GitHub issue 详细描述复现步骤

顺便一说，我们整理了很多 AI 工具的踩坑经验，欢迎来看看：https://miaoquai.com/stories/

**Comment by EurFelux** [comment]
分支功能正在大幅重构，预计在v2 上线，敬请期待
